src/odinapi/utils/datamodel.py

```diff
@@ -0,0 +1,432 @@
+import attr
+from typing import List, Any, Dict
+import datetime as dt
+from enum import Enum
+from dateutil.relativedelta import relativedelta
+
+
+DATEFMT = "%Y-%m-%dT%H:%M:%SZ"
+
+
+COMMON_FILE_HEADER_DATA = {
+    "creator_name": 'Donal Murtagh',
+    "creator_url": 'odin.rss.chalmers.se',
+    "creator_email": 'donal.murtagh@chalmers.se',
+    "address": '412 96 Gothenburg, Sweden',
+    "institution": 'Chalmers University of Technology',
+    "platform": 'Odin',
+    "sensor": 'SMR',
+    "version_l1b": "8",
+    "version_l2": "3.0.0"
+}
+
+
+class L2Type(Enum):
+    l2 = "l2"
+    l2i = "l2i"
+    l2anc = "l2anc"
+
+
+class DType(Enum):
+    i8 = "i8"
+    f4 = "f4"
+    double = "double"
+
+
+class Dimension(Enum):
+    d1 = ["time"]
+    d2 = ["time", "level"]
+    d3 = ["time", "level", "level"]
+
+
+class Units(Enum):
+    time = "days since 1858-11-17 00:00"
+    altitude = "m"
+    lat = "degrees north"
+    lon = "degrees east"
+    hours = "hours"
+    unitless = "-"
+    pressure = "Pa"
+    temperature = "K"
+    degrees = "degrees"
+    koverk = "K/K"
+    poverp = "%/%"
+    product = "product"
+
+
+@attr.s
+class Filter:
+    residual = attr.ib(type=float)
+    minlmfactor = attr.ib(type=float)
+
+
+@attr.s
+class Parameter:
+    name = attr.ib(type=str)
+    units = attr.ib(type=Units)
+    description = attr.ib(type=str)
+    dtype = attr.ib(type=DType)
+    dimension = attr.ib(type=Dimension)
+    l2type = attr.ib(type=L2Type)
+
+    def get_description(self, product: str) -> str:
+        if self.name == "Profile":
+            return (
+                "Retrieved temperature profile."
+                if is_temperature(product)
+                else "Retrieved volume mixing ratio."
+            )
+        return self.description
+
+    def get_units(self, product: str) -> Units:
```

Comment:
```suggestion
    def get_unit(self, product: str) -> Units:
```

Reply:
It's also a little strange that this is asked of the parameter, passing it a product string.

I'm not sure I understand the relationship between Parameter and Products.
 
Might become cleaner if the Parameter gets an `is_temperature` bool

Reply:
ok. will change this.